mrmustard/physics/gaussian_integrals.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains gaussian integral functions and related helper functions.
+"""
+from typing import Sequence, Tuple
+
+import numpy as np
+
+from mrmustard import math
+from mrmustard.utils.typing import ComplexMatrix, ComplexVector
+
+
+def real_gaussian_integral(
+    Abc: tuple,
+    idx: tuple[int, ...],
+):
+    r"""Computes the Gaussian integral of the exponential of a real quadratic form.
+    The integral is defined as (note that in general we integrate over a subset of m dimensions):
+
+    .. :math::
+        \int_{R^m} F(x) dx,
+
+    where
+
+    :math:`F(x) = \textrm{exp}(0.5 x^T A x + b^T x)`
+
+    Here, ``x`` is an ``n``-dim real vector, ``A`` is an ``n x n`` real matrix,
+    ``b`` is an ``n``-dim real vector, ``c`` is a real scalar. The integral indices
+    are specified by ``idx``.
+
+    Arguments:
+        A,b,c: the ``(A,b,c)`` triple
+        idx: the tuple of indices of the x variables
+
+    Returns:
+        The ``(A,b,c)`` triple of the result of the integral.
+    """
+    A, b, c = Abc
+
+    n = len(idx)
```

Comment:
You're only using this in line 76. I would remove it, and replace line 76 with `* math.sqrt((2 * np.pi) ** len(idx) / math.det(M))`